src/data-mgt/python/devices-tranformation/transformation.py

```diff
@@ -17,6 +17,29 @@ def __init__(self, output_format):
         self.airqo_api = AirQoApi()
         self.tahmo_api = TahmoApi()
 
+    def approximate_site_coordinates(self, tenant):
+
+        sites = self.airqo_api.get_sites(tenant)
+        updated_sites = []
+
+        for site in sites:
+
+            try:
+                approximated_coordinates = self.airqo_api.approximate_coordinates(latitude=site["latitude"],
+                                                                                  longitude=site["longitude"],)
+
+                if approximated_coordinates:
+                    updated_sites.append({
+                        "id": site["_id"],
+                        "approximate_latitude": approximated_coordinates["approximate_latitude"],
+                        "approximate_longitude": approximated_coordinates["approximate_longitude"],
+                        "tenant": tenant
+                    })
+            except Exception as ex:
+                print(ex)
+
+        self.airqo_api.update_sites(updated_sites)
```

Comment:
functionality to loop through all current devices and attached approximated coordinates. @Baalmart

Reply:
Alright, looks good to me. Thanks @NoahNsimbe